core/src/main/scala/org/apache/spark/deploy/mesos/MesosDriverDescription.scala

```diff
@@ -40,24 +41,28 @@ private[spark] class MesosDriverDescription(
     val cores: Double,
     val supervise: Boolean,
     val command: Command,
-    val schedulerProperties: Map[String, String],
+    schedulerProperties: Map[String, String],
```

Comment:
Oh right, this is what causes it to become a field. Yeah I suspect lots and lots of things in the code should not even be `val`.